src/scripts/main.js

```diff
@@ -1,3 +1,273 @@
+
+        squares[i].innerHTML = newRow[0];
+        squares[i + 1].innerHTML = newRow[1];
+        squares[i + 2].innerHTML = newRow[2];
+        squares[i + 3].innerHTML = newRow[3];
+      }
+    }
+  }
+
+  function moveUp() {
+    for (let i = 0; i < 4; i++) {
+      const totalOne = squares[i].innerHTML;
+      const totalTwo = squares[i + width].innerHTML;
+      const totalThree = squares[i + (width * 2)].innerHTML;
+      const totalFour = squares[i + (width * 3)].innerHTML;
+      const column = [parseInt(totalOne), parseInt(totalTwo), parseInt(totalThree), parseInt(totalFour)];
+
+      const filteredColumn = column.filter(num => num);
+      const missing = 4 - filteredColumn.length;
+      const zeros = Array(missing).fill(0);
+      const newColumn = filteredColumn.concat(zeros);
+
+      squares[i].innerHTML = newColumn[0];
+      squares[i + width].innerHTML = newColumn[1];
+      squares[i + (width * 2)].innerHTML = newColumn[2];
+      squares[i + (width * 3)].innerHTML = newColumn[3];
+    }
+  }
+
+  function moveDown() {
+    for (let i = 0; i < 4; i++) {
+      const totalOne = squares[i].innerHTML;
+      const totalTwo = squares[i + width].innerHTML;
+      const totalThree = squares[i + (width * 2)].innerHTML;
+      const totalFour = squares[i + (width * 3)].innerHTML;
+      const column = [parseInt(totalOne), parseInt(totalTwo), parseInt(totalThree), parseInt(totalFour)];
+
+      const filteredColumn = column.filter(num => num);
+      const missing = 4 - filteredColumn.length;
+      const zeros = Array(missing).fill(0);
+      const newColumn = zeros.concat(filteredColumn);
+
+      squares[i].innerHTML = newColumn[0];
+      squares[i + width].innerHTML = newColumn[1];
+      squares[i + (width * 2)].innerHTML = newColumn[2];
+      squares[i + (width * 3)].innerHTML = newColumn[3];
+    }
+  }
+
+  function combineRow() {
+    for (let i = 0; i < 15; i++) {
+      if (squares[i].innerHTML === squares[i + 1].innerHTML) {
+        const combinedTotal = parseInt(squares[i].innerHTML) + parseInt(squares[i + 1].innerHTML);
+
+        squares[i].innerHTML = combinedTotal;
+        squares[i + 1].innerHTML = '0';
+        score += combinedTotal;
+        scoreDisplay.innerHTML = score;
+      }
+    }
+    checkForWin();
+  }
+
+  function combineColumn() {
+    for (let i = 0; i < 12; i++) {
+      if (squares[i].innerHTML === squares[i + width].innerHTML) {
+        const combinedTotal = parseInt(squares[i].innerHTML) + parseInt(squares[i + width].innerHTML);
+
+        squares[i].innerHTML = combinedTotal;
+        squares[i + width].innerHTML = '0';
+        score += combinedTotal;
+        scoreDisplay.innerHTML = score;
+      }
+    }
+    checkForWin();
+  }
+
+  function control(e) {
+    if (e.keyCode === 37) {
+      keyLeft();
+    } else if (e.keyCode === 38) {
+      keyUp();
+    } else if (e.keyCode === 39) {
+      keyRight();
+    } else if (e.keyCode === 40) {
+      keyDown();
+    }
+  }
```

Comment:
it's better to use switch case structure here

Reply:
<img width="669" alt="image" src="https://github.com/mate-academy/js_2048_game/assets/70220667/4f1d5aa8-d047-471a-b276-88d6e595ac67">